pgx/src/spi.rs

```diff
@@ -37,25 +37,61 @@ pub enum SpiOk {
     TdRegister = 17,
 }
 
-#[derive(Debug, Primitive)]
+#[derive(Debug, PartialEq)]
+#[repr(i32)]
 pub enum SpiError {
-    // NB:  These are #define'd as negative, but we redefine them as positive so that
-    // #[derive(Primitive)] will work.  We just need to negate result codes from the
-    // various SPI_xxx functions when looking for errors
-    Connect = 1,
-    Copy = 2,
-    Opunknown = 3,
-    Unconnected = 4,
+    Null = 0,
+    Connect = -1,
+    Copy = -2,
+    OpUnknown = -3,
+    Unconnected = -4,
     #[allow(dead_code)]
-    Cursor = 5, /* not used anymore */
-    Argument = 6,
-    Param = 7,
-    Transaction = 8,
-    Noattribute = 9,
-    Nooutfunc = 10,
-    Typunknown = 11,
-    RelDuplicate = 12,
-    RelNotFound = 13,
+    Cursor = -5, /* not used anymore */
+    Argument = -6,
+    Param = -7,
+    Transaction = -8,
+    NoAttribute = -9,
+    NoOutFunc = -10,
+    TypUnknown = -11,
+    RelDuplicate = -12,
+    RelNotFound = -13,
```

Comment:
This might seem like a huge break but it's only an actual functional break if they specifically cast to an integer and then match on it. As-is it only breaks on the ones I gave CamelCase because I don't think there's any actual reason not to. I guess we could keep them as-is...?

Reply:
Well, considering 0.5.0 is going to be breaking anyways, my opinion might be that doing whatever feels more natural (and also reducing dependencies) would be a really good thing. 

What are the integer values used for? Is it only for matching error results? Are the primitives needed for some kind of postgres functionality?

Reply:
Assigning integer values enforces that these values are, and allows you to cast `result_enum as i32`.